### PR TITLE
Initial project setup

### PR DESCRIPTION
**Pull request by sherryhli**
Initial project setup with "minimum runnable" components (in dev mode).

* PostgreSQL and MongoDB database connections configured and tested with a trivial `Person` model (to be removed)
* `create-react-app` TypeScript template
* Containerized every component

Directory structure:
```
.
├── README.md
├── backend
│   ├── python
│   │   ├── Dockerfile
│   │   ├── app
│   │   │   ├── __init__.py
│   │   │   ├── config.py
│   │   │   ├── models
│   │   │   │   ├── __init__.py
│   │   │   │   └── person.py
│   │   │   ├── rest
│   │   │   │   └── placeholder.py
│   │   │   └── services
│   │   │       └── placeholder.py
│   │   ├── requirements.txt
│   │   └── server.py
│   └── typescript
│       ├── Dockerfile
│       ├── graphql
│       │   └── placeholder.ts
│       ├── models
│       │   ├── index.ts
│       │   ├── person.mgmodel.ts
│       │   └── person.pgmodel.ts
│       ├── nodemon.json
│       ├── package.json
│       ├── rest
│       │   └── placeholder.ts
│       ├── server.ts
│       ├── services
│       │   └── placeholder.ts
│       ├── tsconfig.json
│       └── yarn.lock
├── docker-compose.yml
└── frontend
    ├── Dockerfile
    ├── README.md
    ├── package.json
    ├── src
    │   ├── App.css
    │   ├── App.test.tsx
    │   ├── App.tsx
    │   ├── index.css
    │   ├── index.tsx
    │   ├── logo.svg
    │   ├── react-app-env.d.ts
    │   ├── reportWebVitals.ts
    │   └── setupTests.ts
    ├── tsconfig.json
    └── yarn.lock
```

### CODE REVIEW

File: backend/typescript/models/person.pgmodel.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { Column, Model, Table } from "sequelize-typescript";
```
Comment (sherryhli) [inline backend/typescript/models/person.pgmodel.ts:1]: Had issues using `sequelize` directly with TypeScript so went with this [package](sequelize-typescript). I recommend reading the docs.

File: backend/typescript/server.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+const app = express();
+app.use(cors());
+app.use(express.json());
+
+// app.get("/", async (_req, res) => {
+//     await PgPerson.create({name: "First Last", email: "first@last.com"});
+//     await MgPerson.create({name: "First Last", email: "first@last.com"});
+//     res.send("Created some records!");
+// });
+
```
Comment (sherryhli) [inline backend/typescript/server.ts:17]: Just a quick way to test the models, keeping it for demo purposes but should be removed soon

File: backend/typescript/tslint.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+    "defaultSeverity": "error",
+    "extends": [
+        "tslint:recommended"
+    ],
+    "jsRules": {},
+    "rules": {
+        "trailing-comma": [ false ]
```
Comment (sherryhli) [inline backend/typescript/tslint.json:8]: Feel free to suggest more rules

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
+version: "3.7"
+
```
Comment (sherryhli) [inline docker-compose.yml:2]: Comment out sections in `services` as needed for now

File: backend/typescript/models/person.pgmodel.ts (hunk 1 above)
Comment (chenrichard10) [reply backend/typescript/models/person.pgmodel.ts:1]: Link doesn't seem to work

Comment (sherryhli) [reply backend/typescript/models/person.pgmodel.ts:1]: Sorry about that! Updated now, please check again

Comment (chenrichard10) [reply backend/typescript/models/person.pgmodel.ts:1]: Works now

File: backend/typescript/package.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    "@types/cors": "^2.8.10",
+    "@types/dotenv": "^8.2.0",
+    "@types/express": "^4.17.11",
+    "@types/mongoose": "^5.10.3",
+    "@types/node": "^14.14.31",
+    "@types/pg": "^7.14.10",
+    "@types/validator": "^13.1.3",
+    "nodemon": "^2.0.7",
+    "ts-node": "^9.1.1",
+    "tslint": "^6.1.3",
```
Comment (AbhiPrasad) [inline backend/typescript/package.json:34]: Can we use eslint instead? (https://github.com/palantir/tslint/issues/4534).

I have an example PR where I switched from tslint to eslint here: https://github.com/uwblueprint/shoe-project/pull/44/files

File: backend/typescript/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,11 @@
+FROM node:14.15.5-slim
+
+WORKDIR /app
+
+COPY package*.json ./
+RUN npm install
```
Comment (AbhiPrasad) [inline backend/typescript/Dockerfile:6]: Can we use yarn here?

File: backend/typescript/models/person.mgmodel.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import mongoose, { Schema, Document } from "mongoose";
+
+export interface Person extends Document {
+    name: string;
+    email: string;
```
Comment (AbhiPrasad) [inline backend/typescript/models/person.mgmodel.ts:5]: Isn't JS usually 2 spaces instead of 4?

File: backend/typescript/tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,70 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
+
+    /* Basic Options */
+    // "incremental": true,                   /* Enable incremental compilation */
+    "target": "es6",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */
```
Comment (AbhiPrasad) [inline backend/typescript/tsconfig.json:7]: if we are are node LTS (> 14.0), can we go on es2020? From the top of my head access to `Promise.allSettled`, `Optional Chaining` and `BigInt` is rlly nice.

File: docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    depends_on:
+      - db
+  db:
+    image: postgres:12-alpine
+    ports:
+      - 5432:5432
+    volumes:
+      - postgres_data:/var/lib/postgresql/data/
+    env_file:
+      - ./backend/typescript/.env
```
Comment (AbhiPrasad) [inline docker-compose.yml:46]: Should we set an `env_file` for the other services as well?

**Review by AbhiPrasad — COMMENTED** [review]
Exciting! Wanted to address the tslint usage but left a couple other comments as well.

**Review by raveen-singh — APPROVED** [review]

File: docker-compose.yml (hunk 9 above)
Comment (sherryhli) [reply docker-compose.yml]: Switched to using a global `.env` file shared by all services (organized by prefixes), I'll leave it to individual teams to decide on how they'll manage secrets

**Review by alexguo8 — APPROVED** [review]

**Review by chenrichard10 — APPROVED** [review]

**Review by carelynntsai — APPROVED** [review]
🍕🍕🍕

**Review by xinhaoz — APPROVED** [review]

**Pull request merged**